infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt

```diff
@@ -42,10 +47,19 @@ internal class SessionEventRepositoryImpl @Inject constructor(
         return session != null
     }
 
-    override suspend fun getCurrentSessionScope(): EventScope =
-        sessionDataCache.eventScope
-            ?: localSessionScope()
-            ?: createSession()
+    override suspend fun getCurrentSessionScope(): EventScope {
+        val cachedScope = sessionDataCache.eventScope
+        if (cachedScope != null) return cachedScope
+
+        val restoredScope = localSessionScope()
+        if (restoredScope != null) {
+            Simber.w("Restored session from DB")
+            return restoredScope
+        }
+
+        Simber.w("Creating new session DB")
```

Comment:
Maybe "Unexpected creation of new session" or something similar :)

Reply:
I don't think that it is unexpected in all cases.